src/App/Pages/Generator/GeneratorPageViewModel.cs

```diff
@@ -91,6 +95,19 @@ public GeneratorPageViewModel()
             RegenerateUsernameCommand = new AsyncCommand(RegenerateUsernameAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
             ToggleForwardedEmailHiddenValueCommand = new AsyncCommand(ToggleForwardedEmailHiddenValueAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
             CopyCommand = new AsyncCommand(CopyAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
+            CloseCommand = new AsyncCommand(CloseAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
```

Comment:
Change the `onException` method, you're using the same as OnSubmit so if the value of `UsernameTypeSelected` is `UsernameType.ForwardedEmailAlias` you're going to show a confusing error to the user if an exception is raised on close. You can just log the exception and that's it